src/Obfuscatories/AbstractBitAwareShiftObfuscator.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace RobThree\HumanoID\Obfuscatories;
+
+abstract class AbstractBitAwareShiftObfuscator extends BasicShiftObfuscator
```

Comment:
The idea here being that if someone wants to get REALLY specific with things then they can.

This abstract class gives them an "integer bit size aware" method to obfuscate IDs. I suspect this would be a very niche use-case but also still quite an obvious one to include. As some users may desire to obfuscate things in a way that can support multiple bit sizes architectures.

On the other hand, even if this feels obvious, maybe it is so niche that it won't be needed as a first-class feature of the library? From that POV I could see us skipping this unless we have a specific implementation we do want to ship/support that will use this?

Reply:
I think having abstract classes are useful when implementing your own integrations. I don't think it fits in the 80% of use cases, but since it doesn't add that much complexity I think it's okay to include it.